pkg/component/worker/config/loader.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2022 k0s authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/k0sproject/k0s/pkg/constant"
+	kubeutil "github.com/k0sproject/k0s/pkg/kubernetes"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/avast/retry-go"
+	"github.com/sirupsen/logrus"
+)
+
+// Loader loads worker configurations from Kubernetes based on the worker
+// profile name.
+type Loader struct {
+	// Used to load the kubeconfig to use when trying to load the worker config.
+	Kubeconfig clientcmd.KubeconfigGetter
+
+	// Optional function used to create the Kubernetes client from a given REST
+	// config.
+	ClientFactory func(*rest.Config) (kubernetes.Interface, error)
+
+	// Logger to use during loading of the worker configuration. Defaults to the
+	// standard logger if nil.
+	Log logrus.FieldLogger
+}
+
+// Load loads the configuration with the given profile name from Kubernetes.
+func (l *Loader) Load(ctx context.Context, profileName string) (*Profile, error) {
+	log := l.Log
+	if log == nil {
```

Comment:
This and the `l.ClientFactory` check below could be in the constructor. Also, it helps to avoid constructions like this:
```
workerConfig, err := (&workerconfig.Loader{
	Kubeconfig: kubernetes.KubeconfigFromFile(kubeletKubeconfigPath),
}).Load(ctx, c.WorkerProfile)
```

Reply:
I'll check out how it looks like if refactored into a free-standing function without a struct. Would probably be a better fit anyways.

Reply:
Done by converting the `Loader` struct into the `LoadProfile` function.